SearchRequestTests testRandomVersionSerialization failure

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3525/testReport/junit/org.elasticsearch.action.search/SearchRequestTests/testRandomVersionSerialization/


```
java.lang.AssertionError: expected:<1557477236737> but was:<1557477236738>
	at __randomizedtesting.SeedInfo.seed([1FCCB2E005F8D472:57C03CA62753C79C]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.elasticsearch.action.search.SearchRequestTests.testRandomVersionSerialization(SearchRequestTests.java:83)
```

Does not reproduce
```
./gradlew :server:test --tests "org.elasticsearch.action.search.SearchRequestTests.testRandomVersionSerialization" -Dtests.seed=1FCCB2E005F8D472 -Dtests.security.manager=true -Dtests.locale=zgh -Dtests.timezone=Africa/Monrovia -Dcompiler.java=12 -Druntime.java=11
```

--- a/server/src/main/java/org/elasticsearch/action/search/SearchRequest.java
+++ b/server/src/main/java/org/elasticsearch/action/search/SearchRequest.java
@@ -286,10 +286,17 @@ public final class SearchRequest extends ActionRequest implements IndicesRequest
      * ensure that the same value, determined by the coordinating node, is used on all nodes involved in the execution of the search
      * request. When created through {@link #crossClusterSearch(SearchRequest, String[], String, long, boolean)}, this method returns
      * the provided current time, otherwise it will return {@link System#currentTimeMillis()}.
-     *
      */
     long getOrCreateAbsoluteStartMillis() {
         return absoluteStartMillis == DEFAULT_ABSOLUTE_START_MILLIS ? System.currentTimeMillis() : absoluteStartMillis;
+    }
+
+    /**
+     * Returns the provided <code>absoluteStartMillis</code> when created through {@link #crossClusterSearch} and
+     * -1 otherwise.
+     */
+    long getAbsoluteStartMillis() {
+        return absoluteStartMillis;
     }
 
     /**

--- a/server/src/test/java/org/elasticsearch/action/search/SearchRequestTests.java
+++ b/server/src/test/java/org/elasticsearch/action/search/SearchRequestTests.java
@@ -80,7 +80,7 @@ public class SearchRequestTests extends AbstractSearchTestCase {
         SearchRequest deserializedRequest = copyWriteable(searchRequest, namedWriteableRegistry, SearchRequest::new, version);
         assertEquals(searchRequest.isCcsMinimizeRoundtrips(), deserializedRequest.isCcsMinimizeRoundtrips());
         assertEquals(searchRequest.getLocalClusterAlias(), deserializedRequest.getLocalClusterAlias());
-        assertEquals(searchRequest.getOrCreateAbsoluteStartMillis(), deserializedRequest.getOrCreateAbsoluteStartMillis());
+        assertEquals(searchRequest.getAbsoluteStartMillis(), deserializedRequest.getAbsoluteStartMillis());
         assertEquals(searchRequest.isFinalReduce(), deserializedRequest.isFinalReduce());
     }
 

